src/cpu.rs

```diff
@@ -527,6 +549,181 @@ impl Cpu {
         self.condition_codes.set_zero(self.registers.a);
         self.condition_codes.set_parity(self.registers.a);
     }
+
+    // arithmetic group
+    fn add(&mut self, val: u8) {
+        // val is value containted in register
+        let reg_a = self.registers.a;
+        let res: u16 = (reg_a as u16).wrapping_add(val as u16);
+        // put result in accumulator
+        self.registers.a = res as u8;
+        // update flags
+        self.condition_codes.set_zero(res as u8);
+        self.condition_codes.set_sign(res as u8);
+        self.condition_codes.set_parity(res as u8);
+        self.condition_codes.set_carry(res > 0xFF);
+        self.condition_codes
+            .set_aux_carry((reg_a & 0x0F) + (val & 0x0F) > 0x0F);
+    }
+
+    fn inr(&mut self, reg: Operand) {
+        let res = match reg {
+            Operand::A => {
+                self.registers.a = self.registers.a.wrapping_add(1);
+                self.registers.a
+            }
+            Operand::B => {
+                self.registers.b = self.registers.b.wrapping_add(1);
+                self.registers.b
+            }
+            Operand::C => {
+                self.registers.c = self.registers.c.wrapping_add(1);
+                self.registers.c
+            }
+            Operand::D => {
+                self.registers.d = self.registers.d.wrapping_add(1);
+                self.registers.d
+            }
+            Operand::E => {
+                self.registers.e = self.registers.e.wrapping_add(1);
+                self.registers.e
+            }
+            Operand::H => {
+                self.registers.h = self.registers.h.wrapping_add(1);
+                self.registers.h
+            }
+            Operand::L => {
+                self.registers.l = self.registers.l.wrapping_add(1);
+                self.registers.l
+            }
+            Operand::M => {
+                let hl = self.registers.get_hl() as usize;
+                self.memory[hl] = self.memory[hl].wrapping_add(1);
+                self.memory[hl]
+            }
+            _ => panic!("INR only accepts registers or a memory location"),
+        };
+        // update flags
+        self.condition_codes.set_zero(res);
+        self.condition_codes.set_sign(res);
+        self.condition_codes.set_parity(res);
+        self.condition_codes
+            .set_aux_carry((res.wrapping_sub(1) & 0x0F) == 0x00);
```

Comment:
Shouldn't this be comparing against 0x0F instead of 0x00?

Reply:
I will look at this again. The carry and aux carry flags very much confuse me with the subtraction instructions.

Reply:
I changed it to (res & 0xF) == 0x0, which is basically the same thing but then I don't have to use the wrapping_sub